pkg/extensions/sync/utils.go

```diff
@@ -331,8 +333,16 @@ func pushSyncedLocalImage(localRepo, tag, localCachePath string,
 		}
 	}
 
-	_, err = imageStore.PutImageManifest(localRepo, tag, ispec.MediaTypeImageManifest, manifestContent)
+	_, err = imageStore.PutImageManifest(localRepo, tag,
+		ispec.MediaTypeImageManifest, manifestContent)
 	if err != nil {
+		if errors.Is(err, zerr.ErrImageLintAnnotations) {
```

Comment:
you can cover this in sync_internal_test

use the test image under test/data/zot-test/, copy it to a temp dir
localcache would be this temp dir
localRepo zot-test
tag 0.0.1

imagestore should be an imagestore with a non nil linter which has mandatory annotations in its config.

should fail on this line.

This is one way, another way is to make a sync test with a regular upstream with a rootDir which has zot-test image
downstream being a zot with a linter config with mandatory annotation, let it sync and assert that the image was not sync, and check log that contains str "couldn't upload manifest because of missing annotations"